src/common/pages/profile-functional.tsx

```diff
@@ -566,16 +645,34 @@ export const Profile = (props: Props) => {
                     return <Redirect to={`/@${account.name}`} />;
                   }
                 }
-
-                if (data !== undefined) {
-                  let entryList = data?.entries;
-                  entryList = entryList.filter(
-                    (item) => item.permlink !== (account as FullAccount)?.profile?.pinned
-                  );
-                  if (pinnedEntry) {
-                    entryList.unshift(pinnedEntry);
+                if (
+                  data !== undefined ||
+                  (section === "trail" &&
+                    dataTrail.entries !== undefined &&
+                    dataTrail.entries.length > 0)
+                ) {
+                  let entryList;
+                  switch (section) {
```

Comment:
For this switch case I do not suggest doing this filtering at the time of rendering. May be we check and unshift the pinned posts at the time of fetching enteries. 

One more suggestion is to create a separate component for Trail just like in above lines different components are returned on different sections. This will have following benefits:

- **Maintaining Code Integrity:** As I can see that no data fetching code is present inside Profile component. For profile posts the entries are fetched in the store and stored there. I understand that we cannot save the trail in store as different profiles would be opened by the user and it will eventually make the store heavy. Therefore I suggest that we go to a separate component and do all the data fetching there. 
- **Clean Code:** Above approach would clean the code. Also improve performance as there would be less conditions to check at the time of render. Since the component could be optimized for performance and would be data driver, multiple checks on dataTrail would not be needed at the time of render. 
- **Better State Management:** Fetching data  and state management of dataTrail entries would be simplified and Profile component will retain its current code without adding much complexity to it.

**Challenges:**
- **Handling loading:** Since loading is managed in main (Profile) component. To keep simple we might avoid rendering another loading component in our new component. We might want to use the current loading mechanism from Profile component with our new component. For this we could pass a function from Profile component to the new child component and vice versa to manage loading.
- **Handling bottom reached:** Same approach as the above loading mechanism could be followed. 
There could be other challenges also. I am happy and available to help anytime if needed. :)

We can see that EntryListContent component is returned at last by default here and is still another component. So I think it would be a better approach to create a new component for Trail as well. It will help avoid complexity in Profile component.

Basically I suggest to move the Trail data management outside of the Profile component and in a child component.

@feruzm And @dkildar . Please provide your thoughts on this.

If this approach is followed it might rule out some of my previous comments for the Profile component. But it would still be helpful as @igormuba could consider those suggestions while implementing the new component.